From 2b5550a19fb211a3fa65493bd4ec823327684f01 Mon Sep 17 00:00:00 2001
In-Reply-To: <xmqq7dpkdup6.fsf@gitster.c.googlers.com>
References: <xmqq7dpkdup6.fsf@gitster.c.googlers.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Wed, 16 Dec 2020 00:37:37 +0100
Subject: [PATCH 1/2] strmap: do not "return" in a void function
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

On SunCC this is a hard error, and since this code first appears in
the v2.30-0-rc0 release it won't build on Solaris SunCC.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 strmap.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/strmap.h b/strmap.h
index c4c104411bf..1e152d832d6 100644
--- a/strmap.h
+++ b/strmap.h
@@ -165,7 +165,7 @@ static inline int strintmap_contains(struct strintmap *map, const char *str)
 
 static inline void strintmap_remove(struct strintmap *map, const char *str)
 {
-	return strmap_remove(&map->map, str, 0);
+	strmap_remove(&map->map, str, 0);
 }
 
 static inline int strintmap_empty(struct strintmap *map)
@@ -249,7 +249,7 @@ static inline int strset_contains(struct strset *set, const char *str)
 
 static inline void strset_remove(struct strset *set, const char *str)
 {
-	return strmap_remove(&set->map, str, 0);
+	strmap_remove(&set->map, str, 0);
 }
 
 static inline int strset_empty(struct strset *set)
-- 
2.29.2.222.g5d2a92d10f8

